### PR TITLE
Cardano multiasset check firmware version

### PR DESCRIPTION
**Pull request by gabrielKerekes**
Creating an internal PR so we can review here.

### CODE REVIEW

File: src/js/core/methods/CardanoSignTransaction.js [hunk 1]
```diff
@@ -7,6 +7,10 @@ import { validatePath } from '../../utils/pathUtils';
 import { addressParametersToProto, validateAddressParameters } from './helpers/cardanoAddressParameters';
 import { transformCertificate } from './helpers/cardanoCertificate';
 import { validateTokenBundle, tokenBundleToProto } from './helpers/cardanoTokens';
+import { ERRORS } from '../../constants';
+import Device from '../../device/Device';
```
Comment (gabrielKerekes) [inline src/js/core/methods/CardanoSignTransaction.js]: This seems to be an unused import.

File: src/js/core/methods/CardanoSignTransaction.js [hunk 2]
```diff
@@ -17,6 +21,13 @@ import type {
 } from '../../types/trezor/protobuf';
 import type { CoreMessage } from '../../types';
 
+// todo: remove when listed firmwares become mandatory for cardanoSignTransaction
+const CardanoSignTransactionFeatures = Object.freeze({
+    SignStakePoolRegistrationAsOwner: ['', '2.3.5'],
+    ValidityIntervalStart: ['0', '2.3.5'],
+    MultiassetOutputs: ['0', '2.3.5'],
+})
```
Comment (gabrielKerekes) [inline src/js/core/methods/CardanoSignTransaction.js]: Missing `;`.

File: src/js/core/methods/CardanoSignTransaction.js [hunk 3]
```diff
@@ -7,6 +7,10 @@ import { validatePath } from '../../utils/pathUtils';
 import { addressParametersToProto, validateAddressParameters } from './helpers/cardanoAddressParameters';
 import { transformCertificate } from './helpers/cardanoCertificate';
 import { validateTokenBundle, tokenBundleToProto } from './helpers/cardanoTokens';
+import { ERRORS } from '../../constants';
+import Device from '../../device/Device';
+import { CERTIFICATE_TYPE } from '../../constants/cardano';
+
```
Comment (gabrielKerekes) [inline src/js/core/methods/CardanoSignTransaction.js:12]: The empty line shouldn't be here.

File: src/js/core/methods/CardanoSignTransaction.js [hunk 4]
```diff
@@ -114,7 +125,35 @@ export default class CardanoSignTransaction extends AbstractMethod {
...
+            if (certificate.type === CERTIFICATE_TYPE.StakePoolRegistration) {
+                this._ensureFeatureIsSupported('SignStakePoolRegistrationAsOwner');
+            }
+        });
+
+        if (params.validity_interval_start != null) {
+            this._ensureFeatureIsSupported('ValidityIntervalStart');
+        }
+
+        params.outputs.map((output) => {
```
Comment (gabrielKerekes) [inline src/js/core/methods/CardanoSignTransaction.js]: Apparently there's a trailing space at the end of line.

**Review by gabrielKerekes — COMMENTED** [review]
Before creating the PR I thought that the features should be constants. Now I see that it's handled via `$Keys<...>`. Nice. It also probably wouldn't work with constants. 

Oh well, I'll at least mention linter errors I'm getting.

**Comment by refi93** [comment]
ported to destination branch, thanks!

**Pull request closed without merging**